src/AppInstallerRepositoryCore/Rest/Schema/IRestClient.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "Microsoft/Schema/Version.h"
+#include <AppInstallerVersions.h>
+#include <vector>
+
+namespace AppInstaller::Repository::Rest::Schema
+{
+    // The common interface used to interact with RestAPI responses.
+    struct IRestClient
+    {
+    virtual ~IRestClient() = default;
+
+    struct PackageInfo
+    {
+        std::string packageIdentifier;
+        std::string packageName;
+        std::string publisher;
+
+        PackageInfo(std::string packageIdentifier, std::string packageName, std::string publisher) 
+        : packageIdentifier(packageIdentifier), packageName(packageName), publisher(publisher) {}
+    };
+
+    // Minimal information retrieved for any search request.
+    struct Package
+    {
+        PackageInfo packageInfo;
+        std::vector<AppInstaller::Utility::VersionAndChannel> versions;
+
+        Package(PackageInfo packageInfo, std::vector<AppInstaller::Utility::VersionAndChannel> versions)
+        : packageInfo(packageInfo), versions(versions) {}
+    };
+
+    struct SearchResult

```

Comment:
>SearchResult [](start = 11, length = 12)

Is there a reason we cannot use SearchResult definition in Repository::SearchResult?

Reply:
Yes, the sqlite index returns different type of SearchResult (with row ids and such). So we have one specific for Rest source result format.

---
In reply to: [594707007](https://github.com/microsoft/winget-cli/pull/765#discussion_r594707007) [](ancestors = 594707007)